### ISSUE TITLE
Are the required field and custom validation supported for dropdown fields?

### ISSUE THREAD

**Issue by ofarganc** [wcm-io/io.wcm.caconfig.editor#9]
We are trying to make a dropdown field mandatory but it does not work.
Example of dropdown with required
`   @Property(label = "Highlight Type", description = "Defines the type of the highlight.", order = 2, property = {
            "required=true",
            "widgetType=dropdown",
            "dropdownOptions=["
                    + "{'value':'thread','description':'Thread'},"
                    + "{'value':'event','description':'Event'},"
                    + "{'value':'video','description':'Video'},"
                    + "{'value':'podcast','description':'Podcast'},"
                    + "{'value':'article','description':'Article'}"
                    + "]"
    })
    String type() default StringUtils.EMPTY;`

Example of dropdown with custom validation class.
    @Property(label = "Highlight Type", description = "Defines the type of the highlight.", order = 2, property = {
            "validation=highlight-type-mandatory",
            "validationMessage=Highlight Type field is mandatory.",
            "widgetType=dropdown",
            "dropdownOptions=["
                    + "{'value':'thread','description':'Thread'},"
                    + "{'value':'event','description':'Event'},"
                    + "{'value':'video','description':'Video'},"
                    + "{'value':'podcast','description':'Podcast'},"
                    + "{'value':'article','description':'Article'}"
                    + "]"
    })
    String type() default StringUtils.EMPTY;

Is this supported or will it be in the future?
Many thanks.

**Comment by stefanseifert** [comment]
no, currently only "text-based input fields" are supported (as of https://wcm.io/caconfig/editor/validation.html, probably not written explicitly enough), that means text field, text area, pathbrowser.

it would be great to extend the validation support for dropdown fields, but it will be a bit tricky due to the combination of AngularJS and Granite UI. PRs would be welcome!

please note there is also a special property "required" to apply a required validation, this is easier then using a custom validation. but iirc it's also currently not supported for dropdown field for same reason.

i will take a not to improve the documentation around this.

**Comment by ofarganc** [comment]
many thanks for the quick reply @stefanseifert ! We'll take a look and see if we can come up with a PR.
